connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -247,6 +248,16 @@ public boolean isSinkConnector(String connName) {
         }
     }
 
+    /*
+     * Retrieves the connector type by connector name
+     *
+     * @return the ConnectorType for the given connector name
+     * @throws ConnectException if the worker does not manage a connector with the given name.
+     */
+    public ConnectorType getConnectorType(String connName) {
+        return isSinkConnector(connName) ? ConnectorType.SINK : ConnectorType.SOURCE;
```

Comment:
I wonder if this should return null if the connector is not known, rather than throw the ConnectException. Really, that shouldn't happen, but where this method is used a `null` response is harmless whereas an exception would cause an error.